serverless/plugin/step_functions.json

```diff
@@ -1123,94 +1126,154 @@
+    },
+    "Next": {
+      "type": "string"
+    },
+    "End": {
+      "enum": [true]
+    },
+    "Comment": {
+      "type": "string"
+    },
+    "OutputPath": {
+      "type": ["string", "null"]
+    },
+    "InputPath": {
+      "type": ["string", "null"]
+    },
+    "ResultPath": {
+      "type": ["string", "null"]
+    },
+    "ItemsPath": {
+      "type": ["string", "null"]
+    },
+    "MaxConcurrency": {
+      "type": "number",
+      "minimum": 0
+    },
+    "Parameters": {
+      "type": ["object", "array"],
+      "items": {
+        "type": "object"
+      }
+    },
+    "ResultSelector": {
+      "type": "object"
+    },
+    "Retry": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "ErrorEquals": {
+            "type": "array",
+            "items": {
+              "$ref": "#/Aws_StateMachine_Errors"
+            }
+          },
+          "IntervalSeconds": {
+            "type": "number",
+            "minimum": 0
+          },
+          "MaxAttempts": {
+            "type": "number",
+            "minimum": 0
+          },
+          "BackoffRate": {
+            "type": "number",
+            "minimum": 0
+          }
+        },
+        "required": ["ErrorEquals"]
+      }
+    },
+    "Catch": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "ErrorEquals": {
+            "type": "array",
+            "items": {
+              "$ref": "#/Aws_StateMachine_Errors"
+            }
+          },
+          "Next": {
+            "type": "string"
+          }
+        },
+        "required": ["ErrorEquals", "Next"]
+      }
+    }
+  },
+  "Aws_StateMachine_States_Map_ItemProcessor": {
     "type": "object",
     "properties": {
-      "Type": {
-        "type": "string",
-        "enum": ["Map"]
-      },
-      "Next": {
-        "type": "string"
-      },
-      "End": {
-        "enum": [true]
-      },
-      "Comment": {
-        "type": "string"
-      },
-      "OutputPath": {
-        "type": ["string", "null"]
-      },
-      "InputPath": {
-        "type": ["string", "null"]
-      },
-      "ResultPath": {
-        "type": ["string", "null"]
-      },
-      "ItemsPath": {
-        "type": ["string", "null"]
-      },
-      "MaxConcurrency": {
-        "type": "number",
-        "minimum": 0
-      },
-      "Iterator": {
-        "$ref": "#/Aws_StateMachine_Definition"
-      },
-      "Parameters": {
-        "type": ["object", "array"],
-        "items": {
-          "type": "object"
-        }
-      },
-      "ResultSelector": {
-        "type": "object"
-      },
-      "Retry": {
-        "type": "array",
-        "items": {
-          "type": "object",
-          "properties": {
-            "ErrorEquals": {
-              "type": "array",
-              "items": {
-                "$ref": "#/Aws_StateMachine_Errors"
+      "$ref": "#/Aws_StateMachine_States_Map_Properties",
```

Comment:
This isn't the way of doing inheritance with a base model. We're supposed to use `allOf` for it.
I tried doing it, but unfortunately it doesn't work, so we'll have to duplicate these properties.
I can do it if you want, let me know.

Reply:
Updated the code as per comments